### PR TITLE
PE-3960: mnemonic wallet generation and login

### PR DESCRIPTION
**Pull request by karlprieb**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/6d6d2qsob2b0o
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/7hlu0eidq3as0

### CODE REVIEW

File: lib/authentication/login/blocs/login_bloc.dart [hunk 1]
```diff
@@ -307,4 +323,68 @@ class LoginBloc extends Bloc<LoginEvent, LoginState> {
...
+
+    Completer<Wallet> completer = event.walletCompleter;
+    Wallet wallet;
+
+    if (!completer.isCompleted) {
+      emit(const LoginGenerateWallet());
+
+      // wait for minimum 3 seconds
+      var results = await Future.wait(
+          [completer.future, Future.delayed(const Duration(seconds: 3))]);
```
Comment (karlprieb) [inline lib/authentication/login/blocs/login_bloc.dart:354]: Why do we wait for minimum 3 seconds? Why 3 seconds?

File: lib/authentication/login/views/login_page.dart [hunk 2]
```diff
@@ -326,21 +358,41 @@ class _PromptWalletViewState extends State<PromptWalletView> {
...
               children: [
+                if (context
+                    .read<ConfigService>()
+                    .config
+                    .enableSeedPhraseLogin) ...[
+                  ArDriveButton(
+                    icon: ArDriveIcons.keypad(
+                        size: 24, color: colors.themeFgDefault),
+                    key: const Key('loginWithSeedPhraseButton'),
+                    text: "Enter Seed Phrase",
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:376]:
```suggestion
                    // TODO: localization
                    text: "Enter Seed Phrase",
```

File: lib/authentication/login/views/login_page.dart [hunk 3]
```diff
@@ -326,21 +358,41 @@ class _PromptWalletViewState extends State<PromptWalletView> {
...
                   keepButtonVisible: true,
                   width: double.maxFinite,
-                  dragAndDropDescription:
-                      appLocalizationsOf(context).dragAndDropDescription,
-                  dragAndDropButtonTitle:
-                      appLocalizationsOf(context).dragAndDropButtonTitle,
+                  dragAndDropDescription: "Select a KeyFile",
+                  // dragAndDropButtonTitle:
+                  //     appLocalizationsOf(context).dragAndDropButtonTitle,
+                  dragAndDropButtonTitle: "Select a KeyFile",
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:395]:
```suggestion
                  // TODO: create dragAndDropText key
                  dragAndDropDescription: "Select a KeyFile",
                  // dragAndDropButtonTitle:
                  //     appLocalizationsOf(context).dragAndDropButtonTitle,
                  dragAndDropButtonTitle: "Select a KeyFile",
```

File: lib/authentication/login/views/login_page.dart [hunk 4]
```diff
@@ -416,75 +468,100 @@ class _PromptWalletViewState extends State<PromptWalletView> {
...
                             style: ArDriveButtonStyle.secondary,
+                            fontStyle: ArDriveTypography.body
+                                .smallBold700(color: colors.themeFgDefault),
                             onPressed: () {
                               context
                                   .read<LoginBloc>()
                                   .add(const AddWalletFromArConnect());
                             },
-                            text: 'ArConnect',
+                            text: 'Login with ArConnect',
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:536]:
```suggestion
                            // TODO: create new localization key
                            text: 'Login with ArConnect',
```

File: lib/authentication/login/views/login_page.dart [hunk 5]
```diff
@@ -416,75 +468,100 @@ class _PromptWalletViewState extends State<PromptWalletView> {
...
-                            .colors
-                            .themeFgMuted,
-                      )),
-                  TextSpan(
-                    text: appLocalizationsOf(context).dontHaveAWallet2Part,
-                    style: ArDriveTypography.body.smallBold().copyWith(
-                          decoration: TextDecoration.underline,
-                        ),
+                    // text: appLocalizationsOf(context).dontHaveAWallet1Part,
+                    text: "New user? Get started here!",
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:553]:
```suggestion
                    // TODO: create new localization key
                    text: "New user? Get started here!",
```

File: lib/authentication/login/views/login_page.dart [hunk 6]
```diff
@@ -535,18 +615,37 @@ class _LoginCard extends StatelessWidget {
...
+            children: [
+              if (showLattice)
+                Positioned(
+                  bottom: 30,
+                  right: 0,
+                  child: SvgPicture.asset(
+                    isDarkMode
+                        ? Resources.images.login.lattice
+                        : Resources.images.login.latticeLight,
+                    // fit: BoxFit.fitHeight,
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:633]: can we remove this comment?

File: lib/authentication/login/views/login_page.dart [hunk 7]
```diff
@@ -1250,3 +1349,1292 @@ class MaxDeviceSizesConstrainedBox extends StatelessWidget {
...
+                alignment: Alignment.topCenter,
+                child: Text(
+                  'Enter Seed Phrase',
+                  style: ArDriveTypography.headline.headline4Regular(),
+                ),
+              ),
+              const SizedBox(height: 8),
+              Text(
+                // appLocalizationsOf(context).createAndConfirmPassword,
+                'Please enter your 12 word seed phrase and separate each word with a space.',
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:1400]:
```suggestion
                // TODO: create new localization key
                'Please enter your 12 word seed phrase and separate each word with a space.',
```

File: lib/authentication/login/views/login_page.dart [hunk 8]
```diff
@@ -1250,3 +1349,1292 @@ class MaxDeviceSizesConstrainedBox extends StatelessWidget {
...
+
+    if (!isValid) {
+      showAnimatedDialog(context,
+          content: ArDriveIconModal(
+            icon: ArDriveIcons.triangle(
+              size: 88,
+              color: ArDriveTheme.of(context).themeData.colors.themeErrorMuted,
+            ),
+            title: appLocalizationsOf(context).error,
+            content:
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:1508]:
```suggestion
            // TODO: create localization key
            content:
```

File: lib/authentication/login/views/login_page.dart [hunk 9]
```diff
@@ -1250,3 +1349,1292 @@ class MaxDeviceSizesConstrainedBox extends StatelessWidget {
...
+    'With Turbo you can pay with a credit card and increase the reliability of your uploads!',
+    'You can download a copy of your keyfile from the Profile menu.',
+    'If you have large drives, you can take a Snapshot to speed up the syncing time.'
+  ];
+  late String _message;
+
+  @override
+  void initState() {
+    super.initState();
+    _message = 'Did you know?\n\n${_messages[0]}';
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:1549]:
```suggestion
    // TODO: create localization key
    _message = 'Did you know?\n\n${_messages[0]}';
```

File: lib/authentication/login/views/login_page.dart [hunk 10]
```diff
@@ -1250,3 +1349,1292 @@ class MaxDeviceSizesConstrainedBox extends StatelessWidget {
...
+      child: _LoginCard(
+        showLattice: true,
+        content: SingleChildScrollView(
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+            children: [
+              const CircularProgressIndicator(),
+              const SizedBox(height: 16),
+              Text(
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:1581]:
```suggestion
              // TODO: create localization key
              Text(
```

File: lib/authentication/login/views/login_page.dart [hunk 11]
```diff
@@ -1250,3 +1349,1292 @@ class MaxDeviceSizesConstrainedBox extends StatelessWidget {
...
+                          color: colors.themeBorderDefault, width: 1),
+                      color: colors.themeBgSurface,
+                      borderRadius: BorderRadius.circular(6)),
+                  padding: const EdgeInsets.all(16),
+                  child: Row(
+                    children: [
+                      ArDriveIcons.info(size: 24, color: colors.themeFgSubtle),
+                      const SizedBox(width: 16),
+                      Expanded(
+                        child: Text(
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:1626]:
```suggestion
                        // TODO: create localization key
                        child: Text(
```

**Review by karlprieb — COMMENTED** [review]
Great PR!
Can you clean up some unused comments and add `// TODO: create/update localization key` for the texts that doesn't have any translation?
Thank you!

File: lib/authentication/login/blocs/login_bloc.dart (hunk 1 above)
Comment (kunstmusik) [reply lib/authentication/login/blocs/login_bloc.dart:354]: This was requested in the Bounty document:

image

**Review by matibat — DISMISSED** [review]
Approving in the name of @karlprieb  😎

**Pull request merged**